rust/signed_doc/src/metadata/doc_type.rs

```diff
@@ -0,0 +1,225 @@
+        let uuid_v4 = UuidV4::try_from(value)?;
+        Ok(DocType(vec![uuid_v4]))
+    }
+}
+
+impl From<Vec<UuidV4>> for DocType {
+    fn from(value: Vec<UuidV4>) -> Self {
+        DocType(value)
+    }
+}
+
+impl Display for DocType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        write!(
+            f,
+            "[{}]",
+            self.0
+                .iter()
+                .map(UuidV4::to_string)
+                .collect::<Vec<_>>()
+                .join(", ")
+        )
+    }
+}
+
+// ; Document Type
+// document_type = [ 1* uuid_v4 ]
+// ; UUIDv4
+// uuid_v4 = #6.37(bytes .size 16)
+impl Decode<'_, DecodeContext<'_>> for DocType {
+    fn decode(
+        d: &mut Decoder, decode_context: &mut DecodeContext,
+    ) -> Result<Self, minicbor::decode::Error> {
+        const CONTEXT: &str = "DocType decoding";
+        let parse_uuid = |d: &mut Decoder| UuidV4::decode(d, &mut CborContext::Tagged);
+
+        match d.datatype()? {
+            minicbor::data::Type::Array => {
+                let len = d.array()?.ok_or_else(|| {
+                    decode_context
+                        .report
+                        .other("Unable to decode array length", CONTEXT);
+                    minicbor::decode::Error::message(format!(
+                        "{CONTEXT}: Unable to decode array length"
+                    ))
+                })?;
+
+                if len == 0 {
+                    decode_context.report.invalid_value(
+                        "array length",
+                        "0",
+                        "must contain at least one UUIDv4",
+                        CONTEXT,
+                    );
+                    return Err(minicbor::decode::Error::message(format!(
+                        "{CONTEXT}: empty array"
+                    )));
+                }
+
+                (0..len)
+                    .map(|_| parse_uuid(d))
+                    .collect::<Result<Vec<_>, _>>()
+                    .map(Self)
+                    .map_err(|e| {
+                        decode_context
+                            .report
+                            .other(&format!("Invalid UUIDv4 in array: {e}"), CONTEXT);
+                        minicbor::decode::Error::message(format!(
+                            "{CONTEXT}: Invalid UUIDv4 in array: {e}"
+                        ))
+                    })
+            },
+            minicbor::data::Type::Tag => {
+                // Handle single tagged UUID
+                match decode_context.conversion_policy {
+                    ConversionPolicy::Accept | ConversionPolicy::Warn => {
+                        if matches!(decode_context.conversion_policy, ConversionPolicy::Warn) {
+                            warn!("{CONTEXT}: Conversion of document type single UUID to type DocType");
+                        }
+
+                        parse_uuid(d).map(|uuid| Self(vec![uuid])).map_err(|e| {
```

Comment:
Conversion is not just changing the single uuid into an array with one uuid,  it must create a proper array of uuid's that match the specification: https://github.com/input-output-hk/catalyst-libs/blob/main/docs/src/architecture/08_concepts/signed_doc/types.md#document-types

So 
* the proposal uuid maps to: [37(h'7808d2bad51140af84e8c0d1625fdfdc')]
* proposal comment maps to: [37(h'b679ded30e7c41ba89f8da62a17898ea'),
37(h'7808d2bad51140af84e8c0d1625fdfdc')]
* proposal submission action maps to: [37(h'5e60e623ad024a1ba1ac406db978ee48'),
37(h'7808d2bad51140af84e8c0d1625fdfdc'),
37(h'78927329cfd94ea19c710e019b126a65')]

Any other UUID can not be mapped and is an error.